### PR TITLE
Improve return types and make the available endpoints explicit

### PR DESCRIPTION
**Pull request by Smitsel**
Changed the return types to the resource object classes.
And made the endpoint calls explicit, since for instance a /methods call only has get and list.
So when calling ``$mollie->methods->`` my IDE does not suggest create or delete.

### CODE REVIEW

File: src/Endpoints/EndpointAbstract.php [hunk 1]
```diff
@@ -55,18 +55,23 @@ private function buildQueryString(array $filters)
...
      * @param string|null|resource|StreamInterface $body
      * @param array $filters
-     * @return object
+     * @return BaseResource
      * @throws ApiException
      */
-    private function rest_create($restResource, $body, array $filters)
+    protected function rest_create($body, array $filters)
     {
+        $encoded = json_encode($body);
```
Comment (willemstuursma) [inline src/Endpoints/EndpointAbstract.php]: There's a Guzzle one that you can use `\GuzzleHttp\json_encode()`

File: phpunit.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
         <directory>tests/</directory>
     </testsuite>
     <logging>
-        <log type="junit" target="build/logs/junit-unittests.xml" logIncompleteSkipped="false"/>
+        <log type="junit" target="build/logs/junit-unittests.xml"/>
```
Comment (willemstuursma) [inline phpunit.xml]: Unrelated change?

**Comment by willemstuursma** [comment]
Minor stuff.

File: phpunit.xml (hunk 2 above)
Comment (Smitsel) [reply phpunit.xml]: Yes, this flag did not exist for me.
But ill remove the change from this scope.

File: src/Endpoints/EndpointAbstract.php (hunk 1 above)
Comment (Smitsel) [reply src/Endpoints/EndpointAbstract.php]: √

**Comment by Smitsel** [comment]
Thanks, i've updated your feedback 👍

File: src/Endpoints/EndpointAbstract.php [hunk 3]
```diff
@@ -55,18 +55,23 @@ private function buildQueryString(array $filters)
...
-     * @return object
+     * @return BaseResource
      * @throws ApiException
      */
-    private function rest_create($restResource, $body, array $filters)
+    protected function rest_create($body, array $filters)
     {
+        $encoded = \GuzzleHttp\json_encode($body);
+
+        if (json_last_error() !== JSON_ERROR_NONE) {
```
Comment (willemstuursma) [inline src/Endpoints/EndpointAbstract.php]: @Smitsel this Guzzle function already throws on failure.
